pkg/loki/modules.go

```diff
@@ -458,17 +458,19 @@ func (t *Loki) initStore() (_ services.Service, err error) {
 		}
 	}
 
+	if asyncStore {
+		t.Cfg.StorageConfig.EnableAsyncStore = true
+		t.Cfg.StorageConfig.AsyncStoreConfig = storage.AsyncStoreCfg{
+			IngesterQuerier:      t.ingesterQuerier,
+			QueryIngestersWithin: calculateAsyncStoreQueryIngestersWithin(t.Cfg.Querier.QueryIngestersWithin, boltdbShipperMinIngesterQueryStoreDuration(t.Cfg)),
+		}
+	}
+
```

Comment:
It feels a little odd to me having an ingesterquerier inside a config object, would it be possible instead to wrap the storage object here in modules.go rather than inside the init function so we don't have to pass the ingester querier at all?

call the storage.NewStore below this but then kind of like before, if asyncStore then replace the Store inside that object with an async store? WDYT?

Reply:
Agreed, this seems much nicer.

Reply:
We will have to split the factory function for initialising ChunkStore and LokiStore like how it was before. Is that what you mean?

Reply:
I guess so. I'm also fine with merging this as is to fix the bug even though I don't like the pattern we're using and then refactor afterwards.

Reply:
after some discussion we have decided to merge this as is and do a follow up to refactor